### DISCUSSION TITLE
Cookies

### DISCUSSION THREAD

**Discussion by fabioziliani** [RobinTail/express-zod-api#710] [Q&A] [answered]
Is it possible to manage cookies? Or is there a better way to manage authentication?

**Comment by RobinTail** [comment]
Hello @fabioziliani ,

In order to manage (read and set) cookies, you might need to connect `cookie-parser` express middleware.
https://expressjs.com/en/api.html#req.cookies

You can achieve it by using `.use()` or `.addExpressMiddleware()` method of `EndpointsFactory`.

After that, `request.cookies` will be populated by the received cookies. In order to handle them (for authentication purposes) you can write a middleware like this:

```ts
const authMw = createMiddleware({
  input: z.object({}),
  middleware: async ({request, response}) => {
    const { cookies } = request; // here are your cookies
    // do something
    response.cookie('anotherOne', 'some value'); // send another cookie
    return { cookies }; // provide cookies to other middlewares and endpoint handlers as options
    // or just return { isAuthenticated: true } // depending on received cookie examination
  }
});
```

so the final assembly would look this way:

```ts
const myFactory = defaultEndpointsFactory
  .use(cookieParser())
  .addMiddleware(authMw);
const myEndpoint = myFactory.build({ /* ... */ });
```

`response.cookie` is also accessible in a custom `ResultHandler`.

If this approach is not enough for your case, please provide more details on what exactly is missing or would be great to have.

**Comment by fabioziliani** [comment]
Hello @RobinTail and thanks for the reply.
It works; consider writing this also in the docs.
Thanks again.

**Comment by RobinTail** [comment, marked as answer]
First-class cookie support added to v28.1.0, @fabioziliani 
https://ez.robintail.cz/v28.1.0/cookies
